sys/net/link_layer/ieee802154/ieee802154_mac.c

```diff
@@ -0,0 +1,186 @@
+#include "net/netdev/ieee802154.h"
```

Comment:
Does it make sense to make the following submodules?

- `ieee802154_mac`
- `ieee802154_phy`
- `ieee802154_netif`

At least the latter would make sense IMHO if someone wants to use PHY/MAC without `netif`.

Reply:
I think so. In fact, this translates to

```Makefile
#Use raw `at86rf2xx` transceiver (with or without HAL)
USEMODULE += at86rf2xx

# Enable IEEE802.15.4 PHY layer support
USEMODULE += ieee802154_phy

# Enable IEEE802.15.4 MAC layer (automatically enables the IEEE802.15.4 PHY)
USEMODULE += ieee802154_mac

# Enable IEEE802.15.4 network interface (automatically enables PHY and MAC).
USEMODULE += ieee802154_netif
```